src/main/java/tetris/Tetromino.java

```diff
@@ -18,88 +19,121 @@
     private List<Square> miniShape = new ArrayList<>();
 
     public Tetromino() {
-        createOShape();
-        createIShape();
-        createJShape();
-        createLShape();
-        createSShape();
-        createZShape();
-        createTShape();
-        createPShape();
-        createUShape();
-        createMiniShape();
+        getAllShapes();
+        getRandomShape();
+        getCurrentShape();
+    }
+
+    public void getAllShapes() {
+        allShapes.put(1, oShape);
+        allShapes.put(2, iShape);
+        allShapes.put(3, jShape);
+        allShapes.put(4, lShape);
+        allShapes.put(5, sShape);
+        allShapes.put(6, zShape);
+        allShapes.put(7, tShape);
+        allShapes.put(8, pShape);
+        allShapes.put(9, uShape);
+        allShapes.put(10, miniShape);
+    }
+
+    public ArrayList<List<Square>> getShapesList() {
+        return shapesList;
+    }
+
+    public List<Square> getRandomShape() {
+        //place all keys into array
+        Object[] keysArray = allShapes.keySet().toArray();
+        //get random key from keysArray
+        Random random = new Random();
+        Object key = keysArray[random.nextInt(keysArray.length)];
+        //add this shape to the list
+        shapesList.add(allShapes.get(key));
+        //return shape value associated with that random key
+        return allShapes.get(key);
+    }
+
+    public List<Square> getCurrentShape() {
+        List<Square> currShape = shapesList.get(0);
+        for (int i = 0; i < allShapes.size(); i++) {
+            if (currShape != allShapes.get(i)) {
+                continue;
+            } else currShape = allShapes.get(i);
+            //create that shape
```

Comment:
i need to add a call to one of the createShape() methods based on what currShape is